testlib.c

```diff
@@ -32,20 +32,35 @@ void querycb(struct confirm_query *q)/*{{{*/
 
 int main(int argc, char **argv)
 {
-	if(!check_permissions()) { exit(EXIT_FAILURE); }
-	if((argc != 3) && (argc != 4)) {
-		printf("usage: %s iface ttl\n", argv[0]);
+	// run: iface ttl ipversion probe_type
+
+	log_init(LOG_EXTRA, "log.txt", 1, 1024*1024*16);
+
+	if(!check_permissions()) exit(EXIT_FAILURE);
+
+	if(argc!=5){
+		printf("usage: %s iface ttl ipversion probe_type\n", argv[0]);
 		exit(EXIT_FAILURE);
 	}
 
 	char *iface = argv[1];
 	int ttl = atoi(argv[2]);
-	int ipversion; // ipversion = 4 to IPv4; ipversion = 6 to IPv6;
-	if(argc == 4){
-		ipversion = atoi(argv[3]);
+	int ipversion = atoi(argv[3]); // 4 for ipv4, 6 for ipv6
+	int probe_type = atoi(argv[4]); // 1 for icmp, 2 for tcp
+
+	if((probe_type!=1) && (probe_type!=2)){
+		printf("unknown probe type\n");
+		exit(EXIT_FAILURE);		
+	}
+
+	if((ipversion!=4) && (ipversion!=6)){
+		printf("unknown ip version\n");
+		exit(EXIT_FAILURE);
 	}
-	else{
-		ipversion = 4;
+
+	if((probe_type==2) && (ipversion==4)){
```

Comment:
`probe_type` should be compared to the defines. another idea is to use the protocol number, as defined in the standards.